pkg/konnector/controllers/cluster/serviceexport/serviceexport_reconcile.go

```diff
@@ -160,6 +161,18 @@ func (r *reconciler) ensureControllers(ctx context.Context, name string, export
 
 	dynamicConsumerClient := dynamicclient.NewForConfigOrDie(r.consumerConfig)
 	dynamicProviderClient := dynamicclient.NewForConfigOrDie(r.providerConfig)
+
+	providerNamespaceUID := ""
```

Comment:
Still trying to wrap my head around this UID. It's used as owner ref eventually on upsynced objects? Wondering about security implications: the provider does not trust the konnector. Does it have to trust that the right owner-ref is set?

Reply:
Yes, it is on upsynced objects. For example, in the `root:provider` kcp workspace, the Foo `test` refers to its cluster namespace `kube-bind-flbkk`.
```
👉 kube-bind $ kubectl ws .
Current workspace is "root:provider".
👉 kube-bind $ kubectl get ns
NAME                      STATUS   AGE
default                   Active   4d20h
kube-bind-flbkk           Active   4d19h
kube-bind-flbkk-default   Active   70m
👉 kube-bind $ kubectl get foo
NAME                   AGE
kube-bind-flbkk-test   70m
👉 kube-bind $ kubectl get foo kube-bind-flbkk-test -oyaml | yq .metadata.ownerReferences
- apiVersion: v1
  kind: Namespace
  name: kube-bind-flbkk
  uid: 3cb1bc1d-dfd0-4121-9801-37abe86f7d87
👉 kube-bind $ kubectl get ns kube-bind-flbkk -oyaml | yq .metadata.uid
3cb1bc1d-dfd0-4121-9801-37abe86f7d87
```

For the security/trust aspect (as listed in item 1 in the [comment below](https://github.com/kube-bind/kube-bind/pull/175#issuecomment-1721074910)), maybe I'm still not following why 'This trust does not exist'. The konnector holds a kubeconfig that is sent from the provider with proper access, no?

Reply:
This is connected to [my comment of different isolation models](https://github.com/kube-bind/kube-bind/pull/175#issuecomment-1721074910) we might want to implement. This "prefixed" mode is by design not well isolated because you need arbitrary create access and watch permissions. We will want other modes which give more security, especially for Kubernetes providers.

When I say "trust into konnector does not exist" I mean that the provider cannot rely on the consumer running the official konnector. It could be a modified versions that tries malicious requests on the provider cluster.

Reply:
I think I had a better understanding now --- we were leveraging the isolation that offered by k8s namespaces, to limit the access of the potentially malicious konnectors.

I guess this answered one of my dumb questions why there are namespaced `APIServiceNamespace` living in the cluster namespaces.

Is the 'multinsinformer' also driven by this security consideration (we want a consumer be only informed about objects in a few namespaces)?